### PR TITLE
Doc nits: callback function typedefs

### PR DESCRIPTION
**Pull request by richsalz**
Enhance find-doc-nits to be better about finding typedefs for
callback functions.  Fix all nits it now finds.  Added some new
typedef names to ssl.h some of which were documented but did not
exist.

Once this is merged, I'd like to see it as part of one of our travis builds.

### CODE REVIEW

File: doc/man3/SSL_CTX_set_verify.pod [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
-                         int (*verify_callback)(int, X509_STORE_CTX *));
- void SSL_set_verify(SSL *s, int mode,
-                     int (*verify_callback)(int, X509_STORE_CTX *));
+ void SSL_CTX_set_verify(SSL_CTX *ctx, int mode, verify_cb verify_callback);
+ void SSL_set_verify(SSL *s, int mode, verify_cb verify_callback);
  void SSL_CTX_set_verify_depth(SSL_CTX *ctx, int depth);
  void SSL_set_verify_depth(SSL *s, int depth);
 
- int verify_callback(int preverify_ok, X509_STORE_CTX *x509_ctx);
+ typedef int (*verify_cb)(int preverify_ok, X509_STORE_CTX *x509_ctx);
```
Comment (kroeckx) [inline doc/man3/SSL_CTX_set_verify.pod]: I can't find either in the public headers?

Comment (kroeckx) [inline doc/man3/SSL_CTX_set_verify.pod]: Oh, it's added in some other file ...

File: include/openssl/ssl.h [hunk 2]
```diff
@@ -262,6 +262,9 @@ typedef int (*custom_ext_parse_cb) (SSL *s, unsigned int ext_type,
                                     const unsigned char *in,
                                     size_t inlen, int *al, void *parse_arg);
 
+/* Typedef for verification callback */
+typedef int (*verify_cb)(int preverify_ok, X509_STORE_CTX *x509_ctx);
```
Comment (kroeckx) [inline include/openssl/ssl.h]: This looks like a too generic name to put in a public header file.

Comment (richsalz) [inline include/openssl/ssl.h]: fair point; how about SSL_verify_cb?

Comment (richsalz) [inline include/openssl/ssl.h]: ping @kroeckx ...

Comment (kroeckx) [inline include/openssl/ssl.h]: I guess the name is fine. I'm just not sure how I feel about making this change.

Comment (richsalz) [inline include/openssl/ssl.h]: it IS API/ABI compatible, as typedef doesn't introduce changes...
you like it.... you really like it...  you feel your eyes are getting heavy... you are ready to press plus-one...  i am your father ...    :)

Comment (levitte) [inline include/openssl/ssl.h]: > you like it.... you really like it... you feel your eyes are getting heavy... you are ready to press plus-one... i am your father ... :)

Apparently, that had some effect on me ;-)

**Comment by richsalz** [comment]
thanks!

**Pull request closed without merging**